Sources/Alert/Alert.swift

```diff
@@ -0,0 +1,366 @@
+                primaryComparison = lhsProvider.title == rhsProvider.title
+
+            case (nil, nil):
+                primaryComparison = true
+
+            case (.some, nil):
+                primaryComparison = false
+
+            case (nil, .some):
+                primaryComparison = false
+        }
+
+        let secondaryComparison: Bool
+
+        switch (lhs.secondaryButtonProvider, rhs.secondaryButtonProvider) {
+            case let (.some(lhsProvider), .some(rhsProvider)):
+                secondaryComparison = lhsProvider.title == rhsProvider.title
+
+            case (nil, nil):
+                secondaryComparison = true
+
+            case (.some, nil):
+                secondaryComparison = false
+
+            case (nil, .some):
+                secondaryComparison = false
+        }
+
+        return primaryComparison && secondaryComparison
+    }
+
+    func hash(into hasher: inout Hasher) {
+        if let primaryButtonProvider {
+            hasher.combine(primaryButtonProvider.title)
+        }
+
+        if let secondaryButtonProvider {
+            hasher.combine(secondaryButtonProvider.title)
+        }
+    }
+
+    init?(
+        primaryButtonProvider: Warp.Alert.ButtonConstructor?,
+        secondaryButtonProvider: Warp.Alert.ButtonConstructor?,
+        colorProvider: ColorProvider
+    ) {
+        if primaryButtonProvider == nil, secondaryButtonProvider == nil {
+            return nil
+        }
+
+        self.primaryButtonProvider = primaryButtonProvider
+        self.secondaryButtonProvider = secondaryButtonProvider
+        self.colorProvider = colorProvider
+    }
+
+    var body: some View {
+        HStack {
+            if let primaryButtonProvider {
+                Warp.Button.createSecondary(
+                    title: primaryButtonProvider.title,
+                    action: primaryButtonProvider.action
+                )
+            }
+
+            if let secondaryButtonProvider {
+                Warp.Button.createTertiary(
+                    title: secondaryButtonProvider.title,
+                    action: secondaryButtonProvider.action
+                )
+            }
+
+            Spacer()
+        }
+    }
+}
+
+private struct UnderlinedLinkModifier: ViewModifier {
+    private var linkColor: Color {
+        Color.blue
+    }
+
+    func body(content: Content) -> some View {
+        if #available(iOS 16.0, *) {
+            content
+                .underline(true, color: linkColor)
+        } else {
+            content
+                .background(lineBackground)
+        }
+    }
+
+    private var lineBackground: some View {
+        linkColor
+            .frame(height: 1)
+            .offset(y: 8)
+    }
+}
+
+extension Warp.AlertStyle {
+    fileprivate func getBackgroundColor(from colorProvider: ColorProvider) -> Color {
+        switch self {
+            case .info:
+                return FinnColors.aqua50
```

Comment:
We should NOT use Brand colors directly inside any components, that way it will not change color if theme is different. In this case use colorProvider.alertInfoBackground

Reply:
Thanks for your comment, how should I use them?
Since they do not exists in the colors right now, can I add them in color provider and then add them here?
Or I have to wait until it's added?